Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+        )
+    }
+    
+    /// Creates a validated parameter section containing only the parameters that exist in each language representation's function signature.
+    ///
+    /// - Parameters:
+    ///   - parameters: The symbol's documented parameters.
+    ///   - signatures: The symbol's containing only the values that exist in that language representation's function signature.
+    ///   - symbolKind: The documented symbol's kind, for use in diagnostics.
+    ///   - hasDocumentedReturnValues: `true` if the symbol has documented its return values; otherwise `false`.
+    /// - Returns: A parameter section variant containing only the parameters that exist in each language representation's function signature.
+    private func makeParametersSectionVariants(
+        _ parameters: [Parameter]?,
+        _ signatures: Signatures,
+        _ symbolKind: SymbolGraph.Symbol.Kind?,
+        hasDocumentedReturnValues: Bool
+    ) -> DocumentationDataVariants<ParametersSection> {
+        guard let parameters = parameters, !parameters.isEmpty else {
+            guard hasDocumentedReturnValues, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) else {
+                // The symbol has no parameter documentation or return value documentation and none should be synthesized.
+                return DocumentationDataVariants(defaultVariantValue: nil)
+            }
+            // Since the symbol documented its return value synthesize an "error" parameter for it.
+            var variants = DocumentationDataVariants<ParametersSection>()
+            for trait in signatures.keys {
+                variants[trait] = ParametersSection(parameters: []) // Add an empty section so that this language doesn't fallback to the Objective-C content.
+            }
+            variants[.objectiveC] = ParametersSection(parameters: [Parameter(name: "error", contents: Self.objcErrorDescription)]) // This parameter is synthesized and doesn't have a source range.
+            return variants
+        }
+        
+        var variants = DocumentationDataVariants<ParametersSection>()
+        var allKnownParameterNames: Set<String> = []
+        var argumentLabelsAndPossibleParameterNames: [String: Set<String>] = [:]
+        // Group the parameters by name to be able to diagnose parameters documented more than once.
+        let parametersByName = [String: [Parameter]](grouping: parameters, by: \.name)
+        
+        for (trait, signature) in signatures {
+            // Collect all language representations's parameter information from the function signatures.
+            let knownParameterNames = Set(signature.parameters.map(\.name))
+            allKnownParameterNames.formUnion(knownParameterNames)
+            for parameter in signature.parameters {
+                if let externalName = parameter.externalName {
+                    argumentLabelsAndPossibleParameterNames[externalName, default: []].insert(parameter.name)
```

Comment:
Assuming the answer is yes, then maybe we can think of a better name for `argumentLabelsAndPossibleParameterNames`? Something like `parameterNamesByExternalName` or something similar that indicates the keys are the external names.

Reply:
Also, how is it possible for more than one parameter to have the same external name? That is, how could the size of a value in `argumentLabelsAndPossibleParameterNames` have a count > 1?

Maybe I'm misunderstanding the meaning of external name?

Reply:
> Also, how is it possible for more than one parameter to have the same external name? That is, how could the size of a value in `argumentLabelsAndPossibleParameterNames` have a count > 1?

For parameters with separate argument label (external name) and parameter name (name / internal name) then the external name is the first one in the declaration. 

For example, this poorly named function as two parameters with the external name "and":

```swift
func doSomething(with first: Int, and second: Int, and third: Int) {}
//               │    ╰─ name     │   ╰─ name      │   ╰─ name
//               ╰─ externalName  ╰─ externalName  ╰─ externalName
```

Reply:
> Assuming the answer is yes, then maybe we can think of a better name for `argumentLabelsAndPossibleParameterNames`? Something like `parameterNamesByExternalName` or something similar that indicates the keys are the external names.

Sure, I can name this whatever is more readable. Do you prefer `yByX` over `xsAndYs` for a `[X: Y]` dictionary?

Reply:
Personally I prefer `yByX`.... so `parameterNamesByExternalName `. I suppose I was confused by "argumentLabel" which I guess is the same as "externalName".

Reply:
I'm shocked Swift allows repeated external names/argument labels; thanks for the example above.

Reply:
I documented that parameters names are unique but external names aren't in https://github.com/apple/swift-docc/pull/776/commits/ef264671245be85daf65bb1ab7a3eb96a6d8a95b